crates/rust-analyzer/src/handlers.rs

```diff
@@ -1560,13 +1560,22 @@ pub(crate) fn handle_semantic_tokens_range(
 pub(crate) fn handle_open_docs(
     snap: GlobalStateSnapshot,
     params: lsp_types::TextDocumentPositionParams,
-) -> Result<Option<lsp_types::Url>> {
+) -> Result<(Option<lsp_types::Url>, Option<lsp_types::Url>)> {
     let _p = profile::span("handle_open_docs");
     let position = from_proto::file_position(&snap, params)?;
 
-    let remote = snap.analysis.external_docs(position)?;
+    let cargo = match snap.workspaces.get(0) {
+        Some(ProjectWorkspace::Cargo { cargo, .. }) => Some(cargo),
+        _ => None,
+    };
```

Comment:
You can fetch the correct workspace similar to how its done here, https://github.com/rust-lang/rust-analyzer/blob/968850d7bf9916b4d2a6492406a10fd562120b48/crates/rust-analyzer/src/cargo_target_spec.rs#L120-L128